tffm/core.py

```diff
@@ -91,22 +97,32 @@ class TFFMCore():
     ----------
     Steffen Rendle, Factorization Machines
         http://www.csie.ntu.edu.tw/~b97053/paper/Rendle2010FM.pdf
+
     """
     def __init__(self, order=2, rank=2, input_type='dense', loss_function=utils.loss_logistic, 
-                optimizer=tf.train.AdamOptimizer(learning_rate=0.01), reg=0, init_std=0.01, 
-                use_diag=False, reweight_reg=False, seed=None):
+                optimizer=tf.train.AdamOptimizer(learning_rate=0.01), reg=0,
+                init_std=0.01, class_weight=None, use_diag=False, reweight_reg=False,
+                seed=None):
         self.order = order
         self.rank = rank
         self.use_diag = use_diag
         self.input_type = input_type
-        self.loss_function = loss_function
         self.optimizer = optimizer
         self.reg = reg
         self.reweight_reg = reweight_reg
         self.init_std = init_std
         self.seed = seed
         self.n_features = None
         self.graph = None
+        # loss is weighted cross entropy, if class_weight is given
+        if class_weight is not None:
```

Comment:
It's a bad idea to have several almost equal loss functions. It's better to use single loss with a signature like (outputs, y, persample_weights) and calculate persample_weights in fit method.
This will require some additional code (like new placeholder for weights), but this approach allows to implement class weights and sample weights and doesn't require several loss_functions

Reply:
Yeah it's true, I was just doing it the lazy way. I'll re-work this so that there's one `loss_xentropy` with a keyword arg of `pos_weight=1`, which just uses `tf.nn.weighted_cross_entropy_with_logits`. Then I'll use lambda functions, i.e.

`self.loss_function = lambda outputs, y: loss_xentropy(outputs, y, pos_weight = self.class_weights)`

Does that seem reasonable?